slang-kotlin/src/test/java/com/sonarsource/slang/kotlin/KotlinParserTest.java

```diff
@@ -76,6 +77,43 @@ public void testLiterals() {
       .isEquivalentTo(slangStatements("554; true; false; null; \"string\"; 'c';"));
   }
 
+  @Test
+  public void testSimpleStringLiterals() {
+    String escapedBackslash = createEscapedString('\\');
+    String escapedSingleQuote = createEscapedString('\'');
+    String escapedDoubleQuote = createEscapedString('\"');
+    String emptyString = createString("");
+    List<Tree> literals = kotlinStatements(escapedBackslash + ";" + escapedSingleQuote + ";" + escapedDoubleQuote + ";" + emptyString + ";");
+    Tree escapedBackslashLiteral = literals.get(0);
+    Tree escapedSingleQuoteLiteral = literals.get(1);
+    Tree escapedDoubleQuoteLiteral = literals.get(2);
+    Tree emptyStringLiteral = literals.get(3);
+    assertTree(escapedBackslashLiteral).isLiteral(escapedBackslash);
+    assertTree(escapedSingleQuoteLiteral).isLiteral(escapedSingleQuote);
+    assertTree(escapedDoubleQuoteLiteral).isLiteral(escapedDoubleQuote);
+    assertTree(emptyStringLiteral).isLiteral(emptyString);
+  }
+
+  @Test
+  public void testComplexStringLiterals() {
+    List<Tree> complexStrings = kotlinStatements("\"identifier ${x}\"; \"block ${1 == 1}\"");
+    Tree stringWithIdentifier = complexStrings.get(0);
+    Tree identifierExpressionContainer = stringWithIdentifier.children().get(1);
+    assertTree(stringWithIdentifier).isInstanceOf(NativeTree.class);
+    assertTree(stringWithIdentifier.children().get(0)).isLiteral("identifier ");
+    assertTree(identifierExpressionContainer).isInstanceOf(NativeTree.class);
+    assertThat(identifierExpressionContainer.children()).hasSize(1);
+    assertTree(identifierExpressionContainer.children().get(0)).isIdentifier("x");
+
+    Tree stringWithBlock = complexStrings.get(1);
```

Comment:
Why not `Tree escapedBackslashLiteral = kotlinStatement("\"block ${1 == 1}\"");`